File: csharp/dotnet6/DotnetStarter.Logic/ValidationResult.cs
using System.Collections.Generic;
using System.Linq;

namespace DotnetStarter.Logic;

public sealed class ValidationResult
{
    private readonly List<string> errors = new();

    public bool IsValid() => !errors.Any();

    public string Errors() => string.Join("\n", errors);

    public void AddError(string error) => errors.Add(error);
}

File: java/junit5/src/main/java/ValidationResult.java
import java.util.Collections;
import java.util.List;

public final class ValidationResult {

  private final boolean isValid;
  private final String error;

  public ValidationResult(boolean isValid) {
    this.isValid = isValid;
    error = null;
  }

  public ValidationResult(boolean isValid, String error) {
    this.isValid = isValid;
    this.error = error;
  }

  public boolean isValid() {
    return isValid;
  }

  public List<String> errors() {
    return error == null ? Collections.emptyList() : Collections.singletonList(error);
  }
}
